packages/x-data-grid/src/utils/keyboardUtils.ts

```diff
@@ -57,11 +57,23 @@ export const isHideMenuKey = (key: React.KeyboardEvent['key']) => isTabKey(key)
 export function isPasteShortcut(event: React.KeyboardEvent) {
   if (
     (event.ctrlKey || event.metaKey) &&
-    event.key.toLowerCase() === 'v' &&
+    event.code === 'KeyV' &&
     !event.shiftKey &&
     !event.altKey
   ) {
     return true;
   }
   return false;
 }
```

Comment:
I guess this could also be simplified

```tsx
export function isPasteShortcut(event: React.KeyboardEvent) {
  return (event.ctrlKey || event.metaKey) && event.code === 'KeyV' && !event.shiftKey && !event.altKey
}
```